packages/Webkul/Admin/tests/Feature/Catalog/ProductTest.php

```diff
@@ -40,64 +40,699 @@
+        'sku'  => $product->sku,
+        'type' => $product->type,
+    ]);
+
+    $this->assertDatabaseHas('product_flat', [
+        'url_key'              => $product->url_key,
+        'type'                 => 'simple',
+        'name'                 => $name,
+        'short_description'    => $shortDescription,
+        'description'          => $description,
+        'price'                => $price,
+        'weight'               => $weight,
+        'locale'               => $locale,
+        'product_id'           => $product->id,
+        'channel'              => $channel,
+    ]);
+});
+
+it('should return the create page of configurable product', function () {
     // Act and Assert
     $this->loginAsAdmin();
 
-    get(route('admin.catalog.products.copy', $product->id))
-        ->assertRedirect(route('admin.catalog.products.edit', $productId = $product->id + 1))
+    postJson(route('admin.catalog.products.store'), [
+        'type'                => 'configurable',
+        'attribute_family_id' => 1,
+        'sku'                 => fake()->slug(),
+    ])
+        ->assertOk()
+        ->assertJsonPath('data.attributes.0.id', 23)
+        ->assertJsonPath('data.attributes.0.code', 'color')
+        ->assertJsonPath('data.attributes.0.options.0.id', 1)
+        ->assertJsonPath('data.attributes.0.options.0.name', 'Red')
+        ->assertJsonPath('data.attributes.1.id', 24)
+        ->assertJsonPath('data.attributes.1.code', 'size')
+        ->assertJsonPath('data.attributes.1.options.0.id', 6)
+        ->assertJsonPath('data.attributes.1.options.0.name', 'S');
+});
+
+it('should return the edit page of configurable product', function () {
+    // Arrange
+    $product = (new ProductFaker())->getConfigurableProductFactory()->create();
+
+    // Act and Asssert
+    $this->loginAsAdmin();
+
+    get(route('admin.catalog.products.edit', $product->id))
+        ->assertOk()
+        ->assertSeeText(trans('admin::app.catalog.products.edit.title'))
+        ->assertSeeText(trans('admin::app.account.edit.back-btn'))
+        ->assertSeeText($product->url_key)
+        ->assertSeeText($product->name)
+        ->assertSeeText($product->short_description)
+        ->assertSeeText($product->description);
+});
+
+it('should update the configurable product', function () {
+    // Arrange
+    $product = (new ProductFaker())->getConfigurableProductFactory()->create();
+
+    // Act and Asssert
+    $this->loginAsAdmin();
+
+    foreach ($product->variants as $variant) {
+        $variants[$variant->id] = [
+            'sku'         => $variant->sku,
+            'name'        => $variant->name,
+            'price'       => $variant->price,
+            'weight'      => $variant->price,
+            'status'      => $variant->status,
+            'color'       => $variant->color,
+            'size'        => $variant->size,
+            'inventories' => [
+                1 => rand(1111, 9999),
+            ],
+        ];
+    }
```

Comment:
Remove this logic and make another test case i.e. it should be able to update the variant